File: example-android-project/app/src/main/java/io/github/gofaith/faithdroid/FViews/FPopupMenu.java
package io.github.gofaith.faithdroid.FViews;

import android.view.MenuItem;
import android.widget.PopupMenu;

import org.json.JSONArray;
import org.json.JSONTokener;

import faithdroid.Faithdroid;
import io.github.gofaith.faithdroid.UI.AttrGettable;
import io.github.gofaith.faithdroid.UI.AttrSettable;
import io.github.gofaith.faithdroid.UI.Toolkit;
import io.github.gofaith.faithdroid.UI.UIController;

public class FPopupMenu extends FView implements AttrSettable, AttrGettable {
    public PopupMenu v;

    public FPopupMenu(UIController controller, FView anchorView) {
        parentController=controller;
        v = new PopupMenu(parentController.activity, anchorView.view);
    }
    @Override
    public String getAttr(String attr) {
        return null;
    }

    @Override
    public void setAttr(String attr, String value) {
        if (value == null) {
            return;
        }
        switch (attr) {
            case "Menus":
                try {
                    JSONArray array = (JSONArray) (new JSONTokener(value).nextValue());
                    Toolkit.parseMenu(parentController, v.getMenu(), array);
                    v.setOnMenuItemClickListener(new PopupMenu.OnMenuItemClickListener() {
                        @Override
                        public boolean onMenuItemClick(MenuItem item) {
                            if (parentController.menuItemsOnClickMap.containsKey(item)) {
                                Faithdroid.triggerEventHandler(parentController.menuItemsOnClickMap.get(item), "");
                                return true;
                            }
                            return false;
                        }
                    });
                } catch (Exception e) {
                    e.printStackTrace();
                }
                break;
            case "Show":
                v.show();
                break;
            case "Dismiss":
                v.dismiss();
                break;
        }
    }
}

File: android/app/src/main/java/io/github/stevenzack/faithdroidapp/FViews/FPopupMenu.kt
package io.github.stevenzack.faithdroidapp.FViews

import androidx.appcompat.widget.PopupMenu
import faithdroid.Faithdroid
import io.github.stevenzack.faithdroidapp.UI.*
import org.json.JSONArray
import org.json.JSONTokener


class FPopupMenu(controller: UIController, anchorView: FView) : FView(), AttrSettable, AttrGettable {
    var v: PopupMenu

    init {
        parentController = controller
        v = PopupMenu(parentController!!.activity, anchorView.view!!)
    }

   override fun getAttr(attr: String): String{
        return ""
    }

   override fun setAttr(attr: String, value: String) {
        if (value == null) {
            return
        }
        when (attr) {
            "Menus" -> try {
                val array = JSONTokener(value).nextValue() as JSONArray
                Toolkit.parseMenu(parentController!!, v.menu, array)
                v.setOnMenuItemClickListener(PopupMenu.OnMenuItemClickListener { item ->
                    if (parentController!!.menuItemsOnClickMap.containsKey(item)) {
                        Faithdroid.triggerEventHandler(parentController!!.menuItemsOnClickMap.get(item), "")
                        return@OnMenuItemClickListener true
                    }
                    false
                })
            } catch (e: Exception) {
                e.printStackTrace()
            }

            "Show" -> v.show()
            "Dismiss" -> v.dismiss()
        }
    }
}
